addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3315,11 +3420,21 @@ class SnippetsMenu extends Component {
                     'drop-near': option['drop-near'] ? canDrop(option['drop-near'].all()) : false,
                     'drop-in': option['drop-in'] ? canDrop(option['drop-in'].all()) : false,
                 };
-                check = (cache[k]['drop-near'] || cache[k]['drop-in']);
+
+                const hasDropNear = cache[k]['drop-near'];
+                const hasDropIn = cache[k]['drop-in'];
+                check = isSnippetStructure ? hasDropIn : hasDropNear || hasDropIn;
             });
 
             snippet.disabled = !check;
         });
+        // Disable snippet groups that contain no enabled snippets.
+        [...this.snippets.values()].filter(snippetGroup => snippetGroup.snippetGroup)
+                .forEach((snippetGroup) => {
+            snippetGroup.disabled = ![...self.snippets.values()].some(snippet =>
```

Comment:
No need for `self` here inside an arrow function.

Reply:
Also, instead of looping through all snippets once again (because we end up doing that a lot, so I think it's a good idea to limit that where we can), we could keep a set of groups to be shown in the previous loop. Something like:

```js
// before the filter l.3387
const enabledSnippetsGroups = new Set();
...
// at the end of the forEach loop
if (check && snippet.group) {
	enabledSnippetsGroups.add(snippet.group);
}
```
and then
```js
.forEach((snippetGroup) => snippetGroup.disabled = !enabledSnippetsGroups.has(snippetGroup.snippetGroup));
```

Reply:
TODO: post merge